### PR TITLE
`azurerm_dashboard_grafana`: add support for SMTP

### PR DESCRIPTION
**Pull request by jiaweitao001**
* Add support for SMTP configuration.

* fix https://github.com/hashicorp/terraform-provider-azurerm/issues/24635

```
--- PASS: TestAccDashboardGrafana_complete (751.00s)
PASS
```

### CODE REVIEW

File: internal/services/dashboard/dashboard_grafana_resource.go [hunk 1]
```diff
@@ -89,6 +107,70 @@ func (r DashboardGrafanaResource) Arguments() map[string]*pluginsdk.Schema {
 			}, false),
 		},
 
+		"configuration": {
```
Comment (katbyte) [inline internal/services/dashboard/dashboard_grafana_resource.go:110]: i don't think we need this block? configuration is implied?  could we remove this and bring up smtp a level?

File: internal/services/dashboard/dashboard_grafana_resource.go [hunk 2]
```diff
@@ -89,6 +107,70 @@ func (r DashboardGrafanaResource) Arguments() map[string]*pluginsdk.Schema {
...
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*schema.Schema{
+					"smtp": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						MaxItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*schema.Schema{
+								"smtp_enabled": {
```
Comment (katbyte) [inline internal/services/dashboard/dashboard_grafana_resource.go:122]:
```suggestion
								"enabled": {
```
smpt is in the block name?

File: internal/services/dashboard/dashboard_grafana_resource.go [hunk 3]
```diff
@@ -89,6 +107,70 @@ func (r DashboardGrafanaResource) Arguments() map[string]*pluginsdk.Schema {
...
+									Type:         pluginsdk.TypeString,
+									Required:     true,
+									ValidateFunc: validation.StringIsNotEmpty,
+								},
+								"password": {
+									Type:      pluginsdk.TypeString,
+									Required:  true,
+									Sensitive: true,
+								},
+								"skip_verify": {
```
Comment (katbyte) [inline internal/services/dashboard/dashboard_grafana_resource.go:148]: this should be
```suggestion
								"verification_skip_enabled": {
```
?

File: internal/services/dashboard/dashboard_grafana_resource_test.go [hunk 4]
```diff
@@ -217,6 +217,17 @@ resource "azurerm_dashboard_grafana" "test" {
...
+    smtp {
+      smtp_enabled     = true
+      host             = "localhost:25"
+      user             = "user"
+      password         = "password"
+      from_address     = "admin@grafana.localhost"
+      from_name        = "Grafana"
+      start_tls_policy = "OpportunisticStartTLS"
+    }
+  }
```
Comment (katbyte) [inline internal/services/dashboard/dashboard_grafana_resource_test.go:230]:
```suggestion
    smtp {
      enabled     = true
      host             = "localhost:25"
      user             = "user"
      password         = "password"
      from_address     = "admin@grafana.localhost"
      from_name        = "Grafana"
      start_tls_policy = "OpportunisticStartTLS"
    }
```

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/dashboard/dashboard_grafana_resource.go (hunk 1 above)
Comment (jiaweitao001) [reply internal/services/dashboard/dashboard_grafana_resource.go:110]: Will do.

**Review by katbyte — CHANGES_REQUESTED** [review]
Looks like we have a test failure
```
------- Stdout: -------
=== RUN   TestAccDashboardGrafana_basic
    testcase.go:120: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # azurerm_dashboard_grafana.test will be updated in-place
          ~ resource "azurerm_dashboard_grafana" "test" {
                id                                     = "/subscriptions/*******/resourceGroups/acctest-rg-240201193403749019/providers/Microsoft.Dashboard/grafana/a-dg-240201193403749019"
                name                                   = "a-dg-240201193403749019"
                # (12 unchanged attributes hidden)
        
              - smtp {
                  - enabled                   = false -> null
                  - verification_skip_enabled = false -> null
                }
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccDashboardGrafana_basic (600.71s)
FAIL
```

**Review by katbyte — CHANGES_REQUESTED** [review]
there is still a test failure?
```
------- Stdout: -------
=== RUN   TestAccDashboardGrafana_update
    testcase.go:120: Step 2/4 error running import: ImportStateVerify attributes not equivalent. Difference is shown below. The - symbol indicates attributes missing after import.
        
          map[string]string{
        - 	"smtp.0.password": "password",
        + 	"smtp.0.password": "",
          }
--- FAIL: TestAccDashboardGrafana_update (632.32s)
FAIL
```

**Comment by jiaweitao001** [comment]
Screenshot 2024-02-09 091515

Tests are passing as they should.

**Review by katbyte — APPROVED** [review]
LGTM ✉️

**Pull request merged**